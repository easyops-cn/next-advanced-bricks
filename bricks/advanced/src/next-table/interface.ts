import type { ColumnType, TablePaginationConfig } from "antd/es/table";
import type {
  SortOrder,
  ExpandableConfig as TableExpandableConfig,
  TableRowSelection,
} from "antd/es/table/interface.js";
import type { UseBrickConf } from "@next-core/types";
import { CSSProperties } from "react";
import type { TableProps } from "antd/es/table";

export interface NextTableProps {
  rowKey?: string;
  columns?: Column[];
  cell?: CellConfig;
  dataSource?: DataSource;
  frontSearch?: boolean;
  pagination?: PaginationType;
  loading?: boolean;
  multiSort?: boolean;
  sort?: Sort | Sort[];
  rowSelection?: RowSelectionType;
  selectedRowKeys?: (string | number)[];
  hiddenColumns?: (string | number)[];
  expandable?: ExpandableType;
  expandedRowKeys?: (string | number)[];
  childrenColumnName?: string;
  rowDraggable?: boolean;
  searchFields?: (string | string[])[];
  size?: TableProps<RecordType>["size"];
  showHeader?: boolean;
  bordered?: boolean;
  scrollConfig?: TableScroll;
  optimizedColumns?: (string | number)[];
  themeVariant?: "default" | "elevo";
}

export type TableScroll = TableProps<RecordType>["scroll"] | false;

export type RecordType = Record<string, any>;

export interface Column
  extends Pick<
    ColumnType<RecordType>,
    | "align"
    | "colSpan"
    | "dataIndex"
    | "ellipsis"
    | "fixed"
    | "rowScope"
    | "showSorterTooltip"
    | "sortDirections"
    | "title"
    | "width"
  > {
  key?: string | number;
  /** 表头自定义展示构件 */
  headerBrick?: WithUseBrick;
  /** 内容自定义展示构件 */
  useBrick?: UseBrickConf;
  /** 记录表格列合并的值的 key */
  cellColSpanKey?: string;
  /** 记录表格行合并的值的 key */
  cellRowSpanKey?: string;
  /** 是否可排序 */
  sortable?: boolean;
  /** 前端搜索时，多列排序优先级，数字越大优先级越高 */
  sortPriority?: number;
  /** 垂直对齐方式 */
  verticalAlign?: CSSProperties["verticalAlign"];
  /** 单元格样式 */
  cellStyle?: CSSProperties;
  /** 头部单元格样式 */
  headerStyle?: CSSProperties;
  cellStatus?: CellStatus;
}

export interface CellStatus {
  dataIndex?: string;
  mapping: CellStatusMap[];
}

export interface CellStatusMap {
  value: unknown;
  leftBorderColor: string;
}

export interface CellConfig extends WithUseBrick {
  header?: WithUseBrick;
}

export interface WithUseBrick {
  useBrick?: UseBrickConf;
}

export interface DataSource {
  list?: RecordType[];
  total?: number;
  page?: number;
  pageSize?: number;
}

interface PaginationConfig
  extends Pick<
    TablePaginationConfig,
    | "disabled"
    | "hideOnSinglePage"
    | "pageSizeOptions"
    | "responsive"
    | "showLessItems"
    | "showQuickJumper"
    | "showSizeChanger"
    | "showTitle"
    | "simple"
    | "size"
  > {
  showTotal?: boolean;
}

export type PaginationType = false | PaginationConfig | undefined;

interface RowSelectionConfig
  extends Pick<
    TableRowSelection<RecordType>,
    | "columnWidth"
    | "fixed"
    | "hideSelectAll"
    | "preserveSelectedRowKeys"
    | "type"
    | "checkStrictly"
  > {
  showSelectInfo?: boolean;
  rowDisabled?: string | boolean;
}

export type RowSelectionType = boolean | RowSelectionConfig | undefined;

interface ExpandableConfig
  extends Pick<
    TableExpandableConfig<RecordType>,
    | "columnWidth"
    | "expandRowByClick"
    | "defaultExpandAllRows"
    | "fixed"
    | "showExpandColumn"
  > {
  expandIconBrick?: {
    useBrick: UseBrickConf;
  };
  expandedRowBrick?: {
    useBrick: UseBrickConf;
  };
  rowExpandable?: string | boolean;
}

export type ExpandableType = boolean | ExpandableConfig | undefined;

export interface Sort {
  columnKey?: string | number;
  order?: SortOrder;
}
