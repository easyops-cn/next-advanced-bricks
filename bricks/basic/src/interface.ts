import type { GeneralIconProps } from "@next-bricks/icons/general-icon";
/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

/* ============== Common Type Start ============== */

/* ============== Button Type Start ============== */

export type ButtonType =
  | "primary"
  | "default"
  | "dashed"
  | "ghost"
  | "link"
  | "text";

export type ComponentSize = "large" | "medium" | "small" | "xs";

/* ============== Button Type End ============== */

/* ============== Link Type Start ============== */

export type LinkType = "link" | "text" | "plain";

/* ============== Link Type End ============== */

/** ============= Popover Type Start ============== */
export type TriggerEvent = "click" | "hover";
/** ============= Popover Type End ============== */
