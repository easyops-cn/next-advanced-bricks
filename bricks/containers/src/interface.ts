/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

export type ComponentSize = "large" | "middle" | "small" | "xs";

/* ============== Common Type End ============== */

/* ============== Drawer Type Start ============== */

export type Placement = "left" | "right" | "top" | "bottom";

/* ============== Drawer Type Start ============== */
