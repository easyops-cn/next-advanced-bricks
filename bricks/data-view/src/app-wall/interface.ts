import React from "react";
import {Group} from "@tweenjs/tween.js";
import { Vector3Tuple } from "three";
interface TrapezoidalObjectData {
     width: number;
     height: number;
     point: Vector3Tuple
}
export interface TrapezoidalObjectProps {
     // objectRef?: React.RefObject<CSS3DObject>;
     trapezoidalTweenRef: React.RefObject<Group>;
     objectData: TrapezoidalObjectData;
     leftBtnName?: string | undefined;
     rightBtnName?: string | undefined;
     leftOnClick?: () => void;
     rightOnClick?: () => void;

 }
