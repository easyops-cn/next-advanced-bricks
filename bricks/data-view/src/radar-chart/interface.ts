type Color = React.CSSProperties["color"];
export type Colors = Color[][];
type NumArr = number[];
export type Axis = {
  x: number;
  y: number;
};
export interface DrawRadarBgOption {
  layer: number;
  n: number;
  r: number;
  ratio: number;
  origin: number[];
  lineWidth?: number;
}
export interface DrawPolygonOptions extends Omit<DrawRadarBgOption, "layer"> {
  fillStyle?: Color | Colors;
  strokeStyle?: string;
}

export interface GradientPolygon extends DrawPolygonOptions {}
export interface DataLineOptions {
  dataPoints: Axis[];
  strokeStyle?: Color;
  lineWidth?: number;
}
export interface DataCircleOptions extends DataLineOptions {
  r?: number;
  fillStyle?: Color;
}
export interface CircleOptions extends Axis {
  originX: number;
  originY: number;
  r: number;
  lineWidth?: number;
  strokeStyle?: Color;
  fillStyle?: Color;
}
export interface DataFillProps {
  dataPoints: Axis[];
  fillStyle?: Color;
}
export interface DataPolyProps {
  dataLineOptions: DataLineOptions;
  dataFillOptions: DataFillProps;
  dataCircleOptions: DataCircleOptions;
}
export interface BaseConfig {
  n: number;
  dataRadiusOfPercent: NumArr;
  dataRadius: NumArr;
  angleArr: NumArr;
  tooltipsContentArr: NumArr;
  radius: number;
  originDataSource: Data[];
}
export interface UserConfig {
  dataFill: DataFill;
  dataCircle: DataCircle;
  dataLine: DataLine;
}
export interface Data {
  value: number;
  maxValue: number;
  name?: string;
  color?: Color;
  percentValue?: number | string;
}
export type DataFill = {
  fillStyle?: Color;
};
export type DataLine = {
  lineWidth?: number;
  strokeStyle?: Color;
};
export type DataCircle = Omit<DataCircleOptions, "dataPoints">;
export interface RadarProps {
  dataSource: Data[];
  height?: number;
  width?: number;
  radius?: number;
  scale?: number;
  value?: number | string;
  dataFill?: DataFill;
  dataCircle?: DataCircle;
  dataLine?: DataLine;
}
