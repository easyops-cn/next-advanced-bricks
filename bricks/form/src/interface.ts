/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

/* ============== Common Type Start ============== */

/* ============== Button Type Start ============== */

export type ButtonType =
  | "primary"
  | "default"
  | "dashed"
  | "ghost"
  | "link"
  | "text";

export type ComponentSize = "large" | "medium" | "small" | "xs";

/* ============== Button Type End ============== */

/* ============== Input Type Start ============== */

export type InputType =
  | "button"
  | "checkbox"
  | "color"
  | "date"
  | "datetime-local"
  | "email"
  | "file"
  | "hidden"
  | "image"
  | "month"
  | "number"
  | "password"
  | "radio"
  | "range"
  | "reset"
  | "search"
  | "submit"
  | "tel"
  | "text"
  | "time"
  | "url"
  | "week";

/* ============== Input Type End ============== */

/* ============== Input Type Start ============== */

export type Layout = "horizontal" | "vertical" | "inline";

/* ============== Input Type End ============== */

/* ============== Radio Type Start ============== */

export type RadioType =
  | "button"
  | "default"
  | "icon"
  | "icon-circle"
  | "icon-square"
  | "custom";
export interface GeneralComplexOption<T = string | number | boolean> {
  label: string;
  value: T;
  [key: string]: any;
}

export declare type GeneralOption =
  | string
  | number
  | boolean
  | GeneralComplexOption
  | Record<string, any>;

export interface GeneralGroupOption {
  label: string;
  options: GeneralComplexOption | GeneralOption;
}

export declare type RadioGroupButtonStyle = "outline" | "solid";

/* ============== Radio Type End ============== */
