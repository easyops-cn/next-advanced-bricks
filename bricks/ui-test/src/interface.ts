import type { GeneralIconProps } from "@next-bricks/icons/general-icon";

export enum NodeType {
  Suite = "suite",
  Block = "block",
  Command = "command",
}

export interface NodeItem {
  type: NodeType;
  children?: NodeItem[];
  label?: string;
  name: string;
  params?: any[];
}

export interface CommandParam {
  label: string;
  type: CommandParamType;
  description?: string;
  required?: boolean;
  default?: unknown;
  enum?: (string | number | EnumItem)[];
}

export interface CommandDoc {
  name: string;
  category: CommandCategory;
  from: CommandFrom;
  chain: CommandChain;
  description?: string;
  params?: CommandParam[];
  overloads?: CommandOverload[];
  keywords?: string[];
  icon?: CommandIcon;
}

export interface CommandOverload {
  label: string;
  params?: CommandParam[];
}

export type CommandCategory = "query" | "action" | "assertion" | "other";
export type CommandFrom = "cypress" | "third-party" | "custom";
export type CommandChain = "parent" | "child" | "dual";

export type CommandParamType =
  | "number"
  | "string"
  | "boolean"
  | "object"
  | "array"
  | "function"
  | "mixed";

export interface EnumItem {
  label: string;
  value: unknown;
}

export type CommandIcon = GeneralIconProps & {
  color?: string;
};

export interface NodeGraphData {
  name: string;
  label?: string;
  instanceId: string;
  params?: any;
  type: NodeType;
  sort?: number;
}

export interface TreeNodeItemData extends NodeGraphData {
  isChainChild?: boolean;
  isSortChange?: boolean;
  displayLabel?: string;
  parent?: NodeGraphData;
  children?: NodeGraphData[];
}

export interface TestTreeData {
  key: string;
  name: string;
  icon: CommandIcon;
  data: TreeNodeItemData;
  children: TestTreeData[];
  labelColor?: string;
}
