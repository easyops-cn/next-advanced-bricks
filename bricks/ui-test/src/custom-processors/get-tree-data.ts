import { customProcessors } from "@next-core/runtime";
import {
  CommandDoc,
  CommandIcon,
  NodeGraphData,
  TestTreeData,
} from "../interface.js";
import { sortBy } from "lodash";

interface GraphEdge {
  in: string;
  out: string;
  out_name: string;
}

export interface GraphData {
  edges: GraphEdge[];
  topic_vertices: NodeGraphData[];
  vertices: NodeGraphData[];
}

function getIcon(
  nodeData: NodeGraphData,
  commandDocList: CommandDoc[]
): CommandIcon {
  let icon: CommandIcon;

  switch (nodeData.type) {
    case "suite":
      icon = {
        lib: "fa",
        icon: "vial",
        color: "var(--palette-purple-6)",
      };
      break;
    case "block":
      icon = {
        lib: "fa",
        icon: "bookmark",
        color: "var(--palette-cyan-6)",
      };
      break;
    default: {
      const doc = commandDocList?.find((item) => item.name === nodeData.name);
      switch (doc?.category) {
        case "query":
          icon = {
            lib: "antd",
            icon: "aim",
            color: "var(--palette-blue-6)",
          };
          break;
        case "action":
          icon = {
            lib: "fa",
            icon: "computer-mouse",
            color: "var(--palette-red-6)",
          };
          break;
        case "assertion":
          icon = {
            lib: "fa",
            icon: "spell-check",
            color: "var(--palette-green-6)",
          };
          break;
        default:
          icon = {
            lib: "fa",
            icon: "wrench",
            color: "var(--palette-yellow-6)",
          };
      }

      if (doc?.icon) {
        icon = {
          color: icon.color,
          ...doc.icon,
        };
      }
    }
  }

  return icon;
}

function getDisplayLabel(nodeItem: NodeGraphData): string {
  const flag =
    nodeItem.flag && nodeItem.flag !== "none" ? `[${nodeItem.flag}] ` : "";

  if (nodeItem.type === "suite") {
    return nodeItem.name;
  }

  if (nodeItem.label) {
    return `${flag}${nodeItem.name}: ${nodeItem.label}`;
  }

  const literalParams = getLiteralParams(nodeItem.params);
  const nodeLabel = `${flag}${nodeItem.name}`;
  return literalParams.length === 0
    ? nodeLabel
    : `${nodeLabel}: ${literalParams.join(", ")}`;
}

function getLiteralParams(params: unknown[]): unknown[] {
  const literalParams: unknown[] = [];
  if (Array.isArray(params)) {
    for (const param of params) {
      if (
        literalParams.length < 3 &&
        ["string", "boolean", "number"].includes(typeof param)
      ) {
        literalParams.push(param);
      } else {
        break;
      }
    }
  }
  return literalParams;
}

export function getTreeData(
  GraphData: GraphData,
  commandDocList: CommandDoc[]
): TestTreeData {
  const {
    topic_vertices: [rootData],
    vertices,
    edges,
  } = GraphData;

  const getChildVertices = (children: TestTreeData[]) => {
    return sortBy(
      vertices.filter((v) =>
        children.find((c) => c.data.instanceId === v.instanceId)
      ),
      "sort"
    );
  };

  const getChildren = (node: NodeGraphData): TestTreeData[] => {
    const relations = edges.filter((item) => item.in === node.instanceId);

    const nodes = relations
      .map((relation) => vertices.find((v) => relation.out === v.instanceId))
      .filter(Boolean) as NodeGraphData[];

    return sortBy(nodes, "sort").map((child) => getNode(child, node));
  };

  function getNode(node: NodeGraphData, parent?: NodeGraphData): TestTreeData {
    const isChainChild = parent?.type === "command" && node.type === "command";
    const children = getChildren(node);
    let nextChildSort = 0;
    if (children.length > 0) {
      const lastChild = children[children.length - 1];
      nextChildSort = (lastChild.data.sort ?? 0) + 1;
      lastChild.data.isLastChild = true;
    }
    for (const child of children) {
      child.data.nextSiblingSort = nextChildSort;
    }
    const displayLabel = getDisplayLabel(node);
    return {
      name: displayLabel,
      key: node.instanceId,
      icon: getIcon(node, commandDocList),
      data: {
        ...node,
        isChainChild,
        parent,
        displayLabel,
        children: getChildVertices(children),
        nextChildSort,
      },
      children,
    };
  }

  return getNode(rootData);
}

customProcessors.define("uiTest.getTreeData", getTreeData);
