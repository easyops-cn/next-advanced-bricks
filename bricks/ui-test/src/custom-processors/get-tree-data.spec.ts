import { describe, test, expect } from "@jest/globals";
import { GraphData, getTreeData } from "./get-tree-data.js";
import { CommandDoc } from "../interface.js";

describe("getTreeData", () => {
  test.each([
    [
      {
        topic_vertices: [
          {
            instanceId: "603a85a03c2c6",
            label: "demo app",
            name: "suite",
            params: null,
            type: "suite",
          },
        ],
        vertices: [
          {
            instanceId: "603a85ccfb595",
            label: "should work",
            name: "it",
            params: null,
            type: "block",
          },
          {
            instanceId: "603ab42f8eae9",
            label: null,
            name: "beforeEach",
            params: null,
            type: "block",
          },
          {
            instanceId: "603a8bb21c8e6",
            label: null,
            name: "code",
            params: ["console.log(1323)"],
            type: "command",
          },
          {
            instanceId: "603aabdd6d6cd",
            label: null,
            name: "get",
            params: [".selector", { timeout: 1e4 }],
            type: "command",
          },
          {
            instanceId: "603ab463e5d3e",
            label: null,
            name: "code",
            params: null,
            type: "command",
          },
          {
            instanceId: "603aac31912bd",
            label: null,
            name: "type",
            params: null,
            type: "command",
          },
          {
            instanceId: "603aac508c916",
            label: null,
            name: "click",
            params: null,
            type: "command",
          },
          {
            instanceId: "603aac508c917",
            label: null,
            name: "should:exist",
            params: null,
            type: "command",
          },
        ],
        edges: [
          {
            out: "603a85ccfb595",
            in: "603a85a03c2c6",
            out_name: "children",
            properties: null,
          },
          {
            out: "603ab42f8eae9",
            in: "603a85a03c2c6",
            out_name: "children",
            properties: null,
          },
          {
            out: "603a8bb21c8e6",
            in: "603a85ccfb595",
            out_name: "children",
            properties: null,
          },
          {
            out: "603aabdd6d6cd",
            in: "603a85ccfb595",
            out_name: "children",
            properties: null,
          },
          {
            out: "603ab463e5d3e",
            in: "603ab42f8eae9",
            out_name: "children",
            properties: null,
          },
          {
            out: "603aac31912bd",
            in: "603aabdd6d6cd",
            out_name: "children",
            properties: null,
          },
          {
            out: "603aac508c916",
            in: "603aabdd6d6cd",
            out_name: "children",
            properties: null,
          },
          {
            out: "603aac508c917",
            in: "603aac508c916",
            out_name: "children",
            properties: null,
          },
        ],
      },
      [
        {
          name: "code",
          category: "other",
          chain: "dual",
          from: "custom",
          params: [
            {
              label: "Source",
              required: true,
              type: "string",
            },
          ],
          icon: {
            lib: "fa",
            icon: "code",
          },
        },
        {
          name: "get",
          description: "获取元素",
          category: "query",
          from: "cypress",
          chain: "parent",
        },
        {
          name: "type",
          description: "输入内容",
          category: "action",
          from: "cypress",
          chain: "child",
        },
        {
          name: "click",
          description: "点击",
          category: "action",
          from: "cypress",
          chain: "child",
        },
        {
          name: "should:exist",
          category: "assertion",
          from: "cypress",
          chain: "child",
        },
      ],
      {
        children: [
          {
            children: [
              {
                children: [],
                data: {
                  instanceId: "603a8bb21c8e6",
                  isChainChild: false,
                  label: null,
                  name: "code",
                  params: ["console.log(1323)"],
                  parent: {
                    instanceId: "603a85ccfb595",
                    label: "should work",
                    name: "it",
                    params: null,
                    type: "block",
                  },
                  type: "command",
                },
                icon: {
                  color: "var(--palette-yellow-6)",
                  icon: "code",
                  lib: "fa",
                },
                key: "603a8bb21c8e6",
                name: "code: console.log(1323)",
              },
              {
                children: [
                  {
                    children: [],
                    data: {
                      instanceId: "603aac31912bd",
                      isChainChild: true,
                      label: null,
                      name: "type",
                      params: null,
                      parent: {
                        instanceId: "603aabdd6d6cd",
                        label: null,
                        name: "get",
                        params: [".selector", { timeout: 1e4 }],
                        type: "command",
                      },
                      type: "command",
                    },
                    icon: {
                      color: "var(--palette-red-6)",
                      lib: "fa",
                      icon: "computer-mouse",
                    },
                    key: "603aac31912bd",
                    name: "type",
                  },
                  {
                    children: [
                      {
                        children: [],
                        data: {
                          instanceId: "603aac508c917",
                          isChainChild: true,
                          label: null,
                          name: "should:exist",
                          params: null,
                          parent: {
                            instanceId: "603aac508c916",
                            label: null,
                            name: "click",
                            params: null,
                            type: "command",
                          },
                          type: "command",
                        },
                        icon: {
                          color: "var(--palette-green-6)",
                          icon: "spell-check",
                          lib: "fa",
                        },
                        key: "603aac508c917",
                        name: "should:exist",
                      },
                    ],
                    data: {
                      instanceId: "603aac508c916",
                      isChainChild: true,
                      label: null,
                      name: "click",
                      params: null,
                      parent: {
                        instanceId: "603aabdd6d6cd",
                        label: null,
                        name: "get",
                        params: [".selector", { timeout: 1e4 }],
                        type: "command",
                      },
                      type: "command",
                    },
                    icon: {
                      color: "var(--palette-red-6)",
                      lib: "fa",
                      icon: "computer-mouse",
                    },
                    key: "603aac508c916",
                    name: "click",
                  },
                ],
                data: {
                  instanceId: "603aabdd6d6cd",
                  isChainChild: false,
                  label: null,
                  name: "get",
                  params: [".selector", { timeout: 1e4 }],
                  parent: {
                    instanceId: "603a85ccfb595",
                    label: "should work",
                    name: "it",
                    params: null,
                    type: "block",
                  },
                  type: "command",
                },
                icon: {
                  color: "var(--palette-blue-6)",
                  icon: "aim",
                  lib: "antd",
                },
                key: "603aabdd6d6cd",
                name: "get: .selector",
              },
            ],
            data: {
              instanceId: "603a85ccfb595",
              isChainChild: false,
              label: "should work",
              name: "it",
              params: null,
              parent: {
                instanceId: "603a85a03c2c6",
                label: "demo app",
                name: "suite",
                params: null,
                type: "suite",
              },
              type: "block",
            },
            icon: {
              color: "var(--palette-cyan-6)",
              icon: "bookmark",
              lib: "fa",
            },
            key: "603a85ccfb595",
            name: "it: should work",
          },
          {
            children: [
              {
                children: [],
                data: {
                  instanceId: "603ab463e5d3e",
                  isChainChild: false,
                  label: null,
                  name: "code",
                  params: null,
                  parent: {
                    instanceId: "603ab42f8eae9",
                    label: null,
                    name: "beforeEach",
                    params: null,
                    type: "block",
                  },
                  type: "command",
                },
                icon: {
                  color: "var(--palette-yellow-6)",
                  icon: "code",
                  lib: "fa",
                },
                key: "603ab463e5d3e",
                name: "code",
              },
            ],
            data: {
              instanceId: "603ab42f8eae9",
              isChainChild: false,
              label: null,
              name: "beforeEach",
              params: null,
              parent: {
                instanceId: "603a85a03c2c6",
                label: "demo app",
                name: "suite",
                params: null,
                type: "suite",
              },
              type: "block",
            },
            icon: {
              color: "var(--palette-cyan-6)",
              icon: "bookmark",
              lib: "fa",
            },
            key: "603ab42f8eae9",
            name: "beforeEach",
          },
        ],
        data: {
          instanceId: "603a85a03c2c6",
          label: "demo app",
          name: "suite",
          params: null,
          type: "suite",
        },
        icon: {
          color: "var(--palette-purple-6)",
          icon: "vial",
          lib: "fa",
        },
        key: "603a85a03c2c6",
        name: "suite",
      },
    ],
  ])("should work", (GraphData, commandDocs, result) => {
    expect(
      getTreeData(GraphData as GraphData, commandDocs as CommandDoc[])
    ).toEqual(result);
  });
});
