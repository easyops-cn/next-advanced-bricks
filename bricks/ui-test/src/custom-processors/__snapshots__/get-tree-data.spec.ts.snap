// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTreeData should work 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [],
          "data": {
            "children": [],
            "displayLabel": "code: console.log(1323)",
            "instanceId": "603a8bb21c8e6",
            "isChainChild": false,
            "label": null,
            "name": "code",
            "nextChildSort": 0,
            "nextSiblingSort": 1,
            "params": [
              "console.log(1323)",
            ],
            "parent": {
              "instanceId": "603a85ccfb595",
              "label": "should work",
              "name": "it",
              "params": null,
              "type": "block",
            },
            "type": "command",
          },
          "icon": {
            "color": "var(--palette-yellow-6)",
            "icon": "code",
            "lib": "fa",
          },
          "key": "603a8bb21c8e6",
          "name": "code: console.log(1323)",
        },
        {
          "children": [
            {
              "children": [],
              "data": {
                "children": [],
                "displayLabel": "type",
                "instanceId": "603aac31912bd",
                "isChainChild": true,
                "label": null,
                "name": "type",
                "nextChildSort": 0,
                "nextSiblingSort": 1,
                "params": null,
                "parent": {
                  "instanceId": "603aabdd6d6cd",
                  "label": null,
                  "name": "get",
                  "params": [
                    ".selector",
                    {
                      "timeout": 10000,
                    },
                  ],
                  "type": "command",
                },
                "type": "command",
              },
              "icon": {
                "color": "var(--palette-red-6)",
                "icon": "computer-mouse",
                "lib": "fa",
              },
              "key": "603aac31912bd",
              "name": "type",
            },
            {
              "children": [
                {
                  "children": [],
                  "data": {
                    "children": [],
                    "displayLabel": "should:exist",
                    "instanceId": "603aac508c917",
                    "isChainChild": true,
                    "isLastChild": true,
                    "label": null,
                    "name": "should:exist",
                    "nextChildSort": 0,
                    "nextSiblingSort": 1,
                    "params": null,
                    "parent": {
                      "instanceId": "603aac508c916",
                      "label": null,
                      "name": "click",
                      "params": null,
                      "type": "command",
                    },
                    "type": "command",
                  },
                  "icon": {
                    "color": "var(--palette-green-6)",
                    "icon": "spell-check",
                    "lib": "fa",
                  },
                  "key": "603aac508c917",
                  "name": "should:exist",
                },
              ],
              "data": {
                "children": [
                  {
                    "instanceId": "603aac508c917",
                    "label": null,
                    "name": "should:exist",
                    "params": null,
                    "type": "command",
                  },
                ],
                "displayLabel": "click",
                "instanceId": "603aac508c916",
                "isChainChild": true,
                "isLastChild": true,
                "label": null,
                "name": "click",
                "nextChildSort": 1,
                "nextSiblingSort": 1,
                "params": null,
                "parent": {
                  "instanceId": "603aabdd6d6cd",
                  "label": null,
                  "name": "get",
                  "params": [
                    ".selector",
                    {
                      "timeout": 10000,
                    },
                  ],
                  "type": "command",
                },
                "type": "command",
              },
              "icon": {
                "color": "var(--palette-red-6)",
                "icon": "computer-mouse",
                "lib": "fa",
              },
              "key": "603aac508c916",
              "name": "click",
            },
          ],
          "data": {
            "children": [
              {
                "instanceId": "603aac31912bd",
                "label": null,
                "name": "type",
                "params": null,
                "type": "command",
              },
              {
                "instanceId": "603aac508c916",
                "label": null,
                "name": "click",
                "params": null,
                "type": "command",
              },
            ],
            "displayLabel": "get: .selector",
            "instanceId": "603aabdd6d6cd",
            "isChainChild": false,
            "isLastChild": true,
            "label": null,
            "name": "get",
            "nextChildSort": 1,
            "nextSiblingSort": 1,
            "params": [
              ".selector",
              {
                "timeout": 10000,
              },
            ],
            "parent": {
              "instanceId": "603a85ccfb595",
              "label": "should work",
              "name": "it",
              "params": null,
              "type": "block",
            },
            "type": "command",
          },
          "icon": {
            "color": "var(--palette-blue-6)",
            "icon": "aim",
            "lib": "antd",
          },
          "key": "603aabdd6d6cd",
          "name": "get: .selector",
        },
      ],
      "data": {
        "children": [
          {
            "instanceId": "603a8bb21c8e6",
            "label": null,
            "name": "code",
            "params": [
              "console.log(1323)",
            ],
            "type": "command",
          },
          {
            "instanceId": "603aabdd6d6cd",
            "label": null,
            "name": "get",
            "params": [
              ".selector",
              {
                "timeout": 10000,
              },
            ],
            "type": "command",
          },
        ],
        "displayLabel": "it: should work",
        "instanceId": "603a85ccfb595",
        "isChainChild": false,
        "label": "should work",
        "name": "it",
        "nextChildSort": 1,
        "nextSiblingSort": 1,
        "params": null,
        "parent": {
          "instanceId": "603a85a03c2c6",
          "label": "demo app",
          "name": "suite",
          "params": null,
          "type": "suite",
        },
        "type": "block",
      },
      "icon": {
        "color": "var(--palette-cyan-6)",
        "icon": "bookmark",
        "lib": "fa",
      },
      "key": "603a85ccfb595",
      "name": "it: should work",
    },
    {
      "children": [
        {
          "children": [],
          "data": {
            "children": [],
            "displayLabel": "code",
            "instanceId": "603ab463e5d3e",
            "isChainChild": false,
            "isLastChild": true,
            "label": null,
            "name": "code",
            "nextChildSort": 0,
            "nextSiblingSort": 1,
            "params": null,
            "parent": {
              "instanceId": "603ab42f8eae9",
              "label": null,
              "name": "beforeEach",
              "params": null,
              "type": "block",
            },
            "type": "command",
          },
          "icon": {
            "color": "var(--palette-yellow-6)",
            "icon": "code",
            "lib": "fa",
          },
          "key": "603ab463e5d3e",
          "name": "code",
        },
      ],
      "data": {
        "children": [
          {
            "instanceId": "603ab463e5d3e",
            "label": null,
            "name": "code",
            "params": null,
            "type": "command",
          },
        ],
        "displayLabel": "beforeEach",
        "instanceId": "603ab42f8eae9",
        "isChainChild": false,
        "isLastChild": true,
        "label": null,
        "name": "beforeEach",
        "nextChildSort": 1,
        "nextSiblingSort": 1,
        "params": null,
        "parent": {
          "instanceId": "603a85a03c2c6",
          "label": "demo app",
          "name": "suite",
          "params": null,
          "type": "suite",
        },
        "type": "block",
      },
      "icon": {
        "color": "var(--palette-cyan-6)",
        "icon": "bookmark",
        "lib": "fa",
      },
      "key": "603ab42f8eae9",
      "name": "beforeEach",
    },
  ],
  "data": {
    "children": [
      {
        "instanceId": "603a85ccfb595",
        "label": "should work",
        "name": "it",
        "params": null,
        "type": "block",
      },
      {
        "instanceId": "603ab42f8eae9",
        "label": null,
        "name": "beforeEach",
        "params": null,
        "type": "block",
      },
    ],
    "displayLabel": "suite",
    "instanceId": "603a85a03c2c6",
    "isChainChild": false,
    "label": "demo app",
    "name": "suite",
    "nextChildSort": 1,
    "params": null,
    "parent": undefined,
    "type": "suite",
  },
  "icon": {
    "color": "var(--palette-purple-6)",
    "icon": "vial",
    "lib": "fa",
  },
  "key": "603a85a03c2c6",
  "name": "suite",
}
`;
